lib/Serialization/Deserialization.cpp

```diff
@@ -7534,7 +7534,26 @@ Decl *handleErrorAndSupplyMissingClassMember(ASTContext &context,
         error.getNumberOfVTableEntries(),
         error.needsFieldOffsetVectorEntry());
   };
-  llvm::handleAllErrors(std::move(error), handleMissingClassMember);
+
+  // Emit the diagnostics/remarks out of the ModularizationError
+  // wrapped in a TypeError (eg. coming from resolveCrossReference),
+  // which is otherwise just dropped but could help better understand
+  // C/C++ interop issues.
+  assert(context.LangOpts.EnableDeserializationRecovery);
+  auto handleModularizationError = [&](ModularizationError &error)
+      -> llvm::Error {
+    if (context.LangOpts.EnableModuleRecoveryRemarks)
```

Comment:
I really like this as an expansion to `EnableModuleRecoveryRemarks`. Just so you know, we may turn it on by default in the future in some contexts. It currently reports some of the expected errors which we want to be silently dropped, but when we do manage to distinguish between the expected and unexpected errors, we could start reporting only the unexpected errors.

Reply:
Good to know about the chance that the flag may be enabled by default.